jax/_src/lax/lax.py

```diff
@@ -2468,6 +2468,33 @@ def _named_call_translation_rule(c, axis_env, in_nodes, name_stack,
 
 xla.call_translations[named_call_p] = _named_call_translation_rule
 
+
+def named_call(
+    fun: Callable[..., Any],
+    *,
+    name: Optional[str] = None,
+) -> Callable[..., Any]:
+  """Wraps a function in a name_scope with the provided name.
+
+  Args:
+    fun: Function to be wrapped. This can be any Callable.
+    name: Optional. The prefix to use to name all sub computations created
+    within the name scope. Use the fun.__name__ if not specified.
+
+  Returns:
+    A version of `fun` that is wrapped in a name_scope.
+  """
+  if name is None:
+    name = fun.__name__
+
+  _, in_tree = jax.tree_flatten(())
+  def named_f(*args, **kwargs):
```

Comment:
nit: Should we `@functools.wraps(fun)` to preserve docstring et al?

Reply:
Done.